File: ui/react/src/action/SetCellsFormat.ts
import Action from "./Action";
import TableAction from "./TableAction";

class SetCellsFormat extends TableAction {
    public rowIndex: number;
    public columnIndex: number;
    public nRows: number;
    public nColumns: number;
    public format: string;

    public static deserialize(input: any) {
        const sheetName: string = input.sheetName;
        const tableName: string = input.tableName;
        const rowIndex: number = input.rowIndex;
        const columnIndex: number = input.columnIndex;
        const nRows: number = input.nRows;
        const nColumns: number = input.nColumns;
        const format: string = input.format;
        return new SetCellsFormat(
            sheetName,
            tableName,
            rowIndex,
            columnIndex,
            nRows,
            nColumns,
            format
        );
    }

    constructor(
        sheetName: string,
        tableName: string,
        rowIndex: number,
        columnIndex: number,
        nRows: number,
        nColumns: number,
        format: string
    ) {
        super(sheetName, tableName);
        this.rowIndex = rowIndex;
        this.columnIndex = columnIndex;
        this.nRows = nRows;
        this.nColumns = nColumns;
        this.format = format;
    }

    public wrapper(): Action {
        const action = new Action();
        action.setCellsFormat = this;
        return action;
    }
}

export default SetCellsFormat;

File: core/src/main/java/com/ctrip/ferriswheel/core/action/SetCellsFormat.java
/*
 * MIT License
 *
 * Copyright (c) 2018-2019 Ctrip.com
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 *
 */

package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.common.action.Action;

/**
 * @author liuhaifeng
 */
public class SetCellsFormat extends TableAction implements Action {
    private int rowIndex;
    private int columnIndex;
    private int nRows;
    private int nColumns;
    private String format;

    public SetCellsFormat() {
    }

    public SetCellsFormat(String sheetName, String tableName,
                          int rowIndex, int columnIndex, int nRows, int nColumns,
                          String format) {
        super(sheetName, tableName);
        this.rowIndex = rowIndex;
        this.columnIndex = columnIndex;
        this.nRows = nRows;
        this.nColumns = nColumns;
        this.format = format;
    }

    public int getRowIndex() {
        return rowIndex;
    }

    public void setRowIndex(int rowIndex) {
        this.rowIndex = rowIndex;
    }

    public int getColumnIndex() {
        return columnIndex;
    }

    public void setColumnIndex(int columnIndex) {
        this.columnIndex = columnIndex;
    }

    public int getnRows() {
        return nRows;
    }

    public void setnRows(int nRows) {
        this.nRows = nRows;
    }

    public int getnColumns() {
        return nColumns;
    }

    public void setnColumns(int nColumns) {
        this.nColumns = nColumns;
    }

    public String getFormat() {
        return format;
    }

    public void setFormat(String format) {
        this.format = format;
    }
}
